core/network/src/p2p/client.rs

```diff
@@ -0,0 +1,22 @@
+use std::sync::Arc;
+use async_trait::async_trait;
+
+pub type AppResponseCallback = Arc<dyn Fn(Vec<u8>) + Send + Sync>;
+#[async_trait]
+#[allow(unused_variables)]
+pub trait Client: Send + Sync {
+    async fn app_request_any(&mut self, request_bytes: Vec<u8>, on_response: AppResponseCallback) {}
+    async fn app_request(&mut self, request_bytes: Vec<u8>) {}
+    async fn app_gossip(&mut self, request_bytes: Vec<u8>) {}
+    async fn app_gossip_specific(&mut self, request_bytes: Vec<u8>) {}
+    async fn cross_chain_app_request(&mut self, request_bytes: Vec<u8>) {}
+    async fn prefix_message(&mut self, request_bytes: Vec<u8>) {}
+}
```

Comment:
Not sure if you meant to have a default implementation for all of these methods. 

It's also likely that you would want to return `Result` for at least some of these.